vllm/v1/core/kv_cache_manager.py

```diff
@@ -271,6 +271,22 @@ def free(self, request: Request) -> None:
             if block.ref_cnt == 0:
                 self.free_block_queue.append(block)
 
+    def get_num_common_prefix_blocks(
+        self,
+        request: Request,
+        num_requests: int,
+    ) -> int:
+        blocks = self.req_to_blocks[request.request_id]
+        num_common_blocks = 0
+        for block in blocks:
+            # FIXME(woosuk): For some reason, sometimes the ref_cnt is greater
+            # than the number of running requests. DEBUG this.
+            if block.ref_cnt >= num_requests:
+                num_common_blocks += 1
```

Comment:
Please check https://github.com/vllm-project/vllm/pull/11565 and see if that solves the issue.

```suggestion
            if block.ref_cnt == len(self.running):
                num_common_blocks += 1
```

Reply:
Thanks for the bug fix. Fixed.